components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubTopicManagerImpl.java

```diff
@@ -128,24 +131,82 @@ private void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String o
+                    }
+                } else if ((responseCode == HttpStatus.SC_CONFLICT && operation.equals(REGISTER)) ||
+                        (responseCode == HttpStatus.SC_NOT_FOUND && operation.equals(DEREGISTER))) {
+                    HttpEntity entity = response.getEntity();
+                    String responseString = "";
+                    WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                            WebSubHubCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
+                            String.valueOf(responseCode), responsePhrase);
+                    if (entity != null) {
+                        responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+                    }
+                    log.warn(String.format(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB.getDescription(),
+                            topic, operation, responseString));
+                } else {
+                    WebSubHubCorrelationLogUtils.triggerCorrelationLogForResponse(httpPost, requestStartTime,
+                            WebSubHubCorrelationLogUtils.RequestStatus.CANCELLED.getStatus(),
+                            String.valueOf(responseCode), responsePhrase);
+                    if (responseCode == HttpStatus.SC_FORBIDDEN) {
+                        Map<String, String> hubResponse = parseEventHubResponse(response);
+                        if (!hubResponse.isEmpty() && hubResponse.containsKey(HUB_REASON)) {
+                            String errorMsg = String.format(ERROR_TOPIC_DEREG_FAILURE_ACTIVE_SUBS, topic);
+                            if (errorMsg.equals(hubResponse.get(HUB_REASON))) {
+                                log.error(String.format(TOPIC_DEREGISTRATION_FAILURE_ACTIVE_SUBS.getDescription(),
+                                        topic, hubResponse.get(HUB_ACTIVE_SUBS)));
+                                throw handleClientException(TOPIC_DEREGISTRATION_FAILURE_ACTIVE_SUBS, topic,
+                                        hubResponse.get(HUB_ACTIVE_SUBS));
+                            }
+                        }
+                    }
+                    HttpEntity entity = response.getEntity();
+                    String responseString = "";
+                    if (entity != null) {
+                        responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+                    }
+                    String message = String.format(ERROR_BACKEND_ERROR_FROM_WEBSUB_HUB.getDescription(),
+                            topic, operation, responseString);
+                    log.error(message + ", Response code:" + responseCode);
+                    throw new WebSubAdapterServerException(message, ERROR_BACKEND_ERROR_FROM_WEBSUB_HUB.getCode());
+                }
+
+                break;
+            } catch (IOException e) {
```

Comment:
It's important to handle connection time outs and connection read time outs and retry

Basically retries should happen at 500 error range, and connection time outs mainly